contentcuration/contentcuration/static/js/edit_channel/models.js

```diff
@@ -87,6 +89,109 @@ var BaseCollection = Backbone.Collection.extend({
+        var self = this;
+        var promise = new Promise(function(resolve, reject){
+            var promises = [];
+            ids.forEach(function(id){
+                promises.push(new Promise(function(modelResolve, modelReject){
+                    var model = self.get({'id' : id});
+                    if(force_fetch || !model){
+                        model = self.add({'id': id});
+                        model.fetch({
+                            success:function(returned){
+                                modelResolve(returned);
+                            },
+                            error:function(obj, error){
+                                modelReject(error);
+                            }
+                        });
+                    } else {
+                        modelResolve(model);
+                    }
+                }));
+            });
+            Promise.all(promises).then(function(fetchedModels){
+                var to_fetch = self.clone();
+                to_fetch.reset();
+                fetchedModels.forEach(function(entry){
+                    to_fetch.add(entry);
+                });
+                resolve(to_fetch);
+            });
+        });
+        return promise;
+    },
+    destroy:function(){
+        var self = this;
+        return new Promise(function(resolve, reject){
+            var promise_list = [];
+            self.forEach(function(model){
+                promise_list.push(new Promise(function(subresolve, subreject){
+                    model.destroy({
+                        success:subresolve,
+                        error:subreject
+                    })
+                }))
+            });
+            Promise.all(promise_list).then(function(){
+                resolve(true);
+            });
+        });
+    },
+    getName: function(){
+        return this.model_name;
+    },
+    parseRecords: function (resp) {
+        // console.log('response', resp)
+        return resp.results;
+    },    
+    parseState: function (resp, queryParams, state) {
+        state.totalRecords = resp.count;
+        state.totalPages = resp.total_pages;
+
+        return state;
+    },
+    state: {
+        pageSize: DEFAULT_ADMIN_PAGE_SIZE,
+        firstPage: 1,
+        currentPage: 1,
+        filterQuery: {},
+        order: -1,
+    },
+    baseQueryParams: {
+        currentPage: "page",
+        pageSize: "page_size",
+        totalRecords: "count",
+        order: null,
+        ordering: function () {
+            var sortKey = this.state.sortKey, order = this.state.order;
+            if (sortKey && order !== 0) {
+                return (order === 1 ? '-' : '') + sortKey;
+            }
+            return null;
+        }
+    },
+    fetch: function(options) {
+        // Construct the queryParams
+        this.queryParams = Object.assign({}, this.baseQueryParams)
+        this.queryParams = Object.assign(this.queryParams, this.state.filterQuery)
```

Comment:
this merges the params from the router, (set in) `this.state.filterQuery`, with the default `baseQueryParams`